regulations/static/regulations/js/source/views/main/main-view.js

```diff
@@ -124,61 +105,87 @@ var MainView = Backbone.View.extend({
+        delete(this.childView);
+      }
 
-        if (typeof options.render === 'undefined') {
-            // tell the child view it should render
-            options.render = true;
-        }
+      this.loading();
+      SidebarEvents.trigger('section:loading');
 
-        options.id = id;
-        options.type = this.contentType;
-        options.regVersion = this.regVersion;
-        options.docId = this.docId;
-        options.model = this.modelmap[this.contentType];
-        options.cb = this.render;
-        options.cfrTitle = this.cfrTitle;
-
-        // diffs need some more version context
-        if (this.contentType === 'diff') {
-            options.baseVersion = this.regVersion || Helpers.findVersion(Resources.versionElements);
-            options.newerVersion = Helpers.findDiffVersion(Resources.versionElements);
-            if (typeof options.fromVersion === 'undefined') {
-                options.fromVersion = $('#table-of-contents').data('from-version');
-            }
-        }
+      this.setChildOptions(_.extend({render: true}, options));
+
+      this.childModel = this.modelmap[this.contentType];
+      this.childModel.get(id, this.childOptions)
+        .then(this.renderView.bind(this))
+        .fail(this.renderError.bind(this));
+    },
 
-        //search needs to know which version to search and switch to that version
-        if (this.contentType === 'search-results' && typeof options.searchVersion !== 'undefined') {
-            options.regVersion = options.searchVersion;
+    setChildOptions: function(options) {
+      this.childOptions = _.extend({
+        id: this.sectionId,
+        type: this.contentType,
+        regVersion: this.regVersion,
+        docId: this.docId,
+        model: this.childModel,
+        cfrTitle: this.cfrTitle
+      }, options);
+
+      // Diffs need some more version context
+      if (this.contentType === 'diff') {
+        this.childOptions.baseVersion = this.regVersion || Helpers.findVersion(Resources.versionElements);
+        this.childOptions.newerVersion = Helpers.findDiffVersion(Resources.versionElements);
+        if (typeof options.fromVersion === 'undefined') {
+          this.childOptions.fromVersion = $('#table-of-contents').data('from-version');
         }
+      }
 
-        this.loading();
-        SidebarEvents.trigger('section:loading');
+      // Search needs to know which version to search and switch to that version
+      if (this.contentType === 'search-results' && typeof options.searchVersion !== 'undefined') {
+        this.childOptions.regVersion = options.searchVersion;
+      }
+    },
 
-        if (typeof this.childView !== 'undefined') {
-            this.childView.remove();
-            delete(this.childView);
-        }
+    renderView: function(html) {
+      if (html) {
+        this.$el.html(html);
+      }
+
+      this.childOptions.el = this.$el.children().get(0);
+      this.childView = new this.viewmap[this.contentType](this.childOptions);
```

Comment:
Curious -- `renderView` and `createView` are both instantiating this `childView`. Why are they separate functions?

Somewhat related, in `createView` any existing `childView` is destroyed; do we need to do the same here?